app/models/keyword_search_entry.rb

```diff
@@ -7,4 +7,6 @@ class KeywordSearchEntry < ApplicationRecord
   belongs_to :keyword
 
   validates :kind, :position, :urls, presence: true
+
+  scope :belongs_to_user, ->(user_id) { joins(:keyword).where(keywords: { user_id: user_id }) }
```

Comment:
It see best to use the gem pundit for that purpose as thiw will get repetitive. See the following:
- Pundit Policy: https://github.com/nimblehq/mia-web/blob/development/app/policies/campaign_policy.rb (especially the scope part which handles the query)
- Controller usage: https://github.com/nimblehq/mia-web/blob/development/engines/mia_campaign/app/controllers/mia_campaign/campaigns_controller.rb#L9